File: product-manager-master/ProductManager/ProductManagerModel/Domain/Purchase.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace ProductManager.Models
{
    /// <summary>
    /// Purchase is record of the purchase of a product.
    /// </summary>
    public class Purchase : BaseEntity
    {
        /// <summary>
        /// The product Id of the product which has purchased.
        /// </summary>
        public long ProductId { get; set; }

        /// <summary>
        /// Seller Id of seller of the purchase.
        /// </summary>
        public int SellerId { get; set; }

        /// <summary>
        /// The product which has purchased.
        /// </summary>
        public virtual Product Product { get; set; }

        /// <summary>
        /// Seller of the purchase. Purchased product whom?
        /// </summary>
        public virtual Seller Seller { get; set; }

        /// <summary>
        /// Amount of the product.
        /// </summary>
        public decimal Amount { get; set; }

        /// <summary>
        /// Cost of the purchase.
        /// </summary>
        public decimal Cost { get; set; }

        /// <summary>
        /// When the purchase happened.
        /// </summary>
        public DateTime DateTime { get; set; }
    }
}

File: product-manager-master/ProductManagerUI/src/app/model/purchase.ts
import {BaseModel} from "./basemodel";
import {Seller} from "./seller";
import {Product} from "./product";

export class Purchase implements BaseModel {

  constructor(
    public id?: number,
    public sellerFirstName?: string,
    public sellerLastName?: string,
    public productName?: string,
    public amount?: number,
    public cost?: number,
    public dateTime?: Date
  ) {
  }
}

export class PurchaseEditDTO implements BaseModel{

  constructor(
    public id?: number,
    public amount?:string,
    public cost?:string,
    public sellerId? : number,
    public productId? : number,
    public dateTime? : string
) {}

}

export class PurchaseSingleResult {

  constructor(
    public id?: number,
    public amount?:string,
    public cost?:string,
    public seller? : Seller,
    public product? : Product,
    public dateTime? : Date
  ) {
  }
}


export class PurchasePagedResult {

  constructor(
    public count: number,
    public purchases: Purchase[],
  ) {
  }
}
